sdk/trace/span.go

```diff
@@ -609,11 +623,11 @@ func startSpanInternal(ctx context.Context, tr *tracer, name string, o *trace.Sp
 	}
 
 	startTime := o.Timestamp()
+	var stopwatch = defaultStopwatch(startTime)
 	if startTime.IsZero() {
-		startTime = time.Now()
+		stopwatch = tr.provider.clock.Stopwatch()
 	}
-	span.startTime = startTime
-
+	span.stopwatch = stopwatch
```

Comment:
Removing `defaultStopwatch` will remove an unnecessary allocation here.

```
	if startTime.IsZero() {
		span.stopwatch = tr.provider.clock.Stopwatch()
	} else {
		span.stopwatch = standardStopwatch(startTime)
	}
```

The other allocation is going to come from storing the new `Stopwatch` interface type in the span AFAIK. Not sure this is going to be avoidable if we want to track that state.